src/OpenLoco/src/World/StationManager.cpp

```diff
@@ -636,6 +636,84 @@ namespace OpenLoco::StationManager
         station->name = StringIds::null;
     }
 
+    // 0x004901B0
+    NearbyStation findNearbyStation(World::Pos3 pos, CompanyId companyId)
+    {
+        const auto tilePosA = World::toTileSpace(pos) - World::TilePos2(2, 2);
+        const auto tilePosB = World::toTileSpace(pos) + World::TilePos2(2, 2);
+
+        auto minDistanceStation = StationId::null;
+        auto minDistance = std::numeric_limits<int16_t>::max();
+        bool isPhysicallyAttached = false;
+        for (const auto tilePos : World::getClampedRange(tilePosA, tilePosB))
+        {
+            const auto tile = World::TileManager::get(tilePos);
+            for (auto& el : tile)
+            {
+                auto* elStation = el.as<World::StationElement>();
+                if (elStation == nullptr)
+                {
+                    continue;
+                }
+                if (elStation->isGhost())
+                {
+                    continue;
+                }
+                auto* station = StationManager::get(elStation->stationId());
+                if (station->owner != companyId)
+                {
+                    continue;
+                }
+                auto distDiff = World::toWorldSpace(tilePos) - pos;
+                distDiff.x = std::abs(distDiff.x);
+                distDiff.y = std::abs(distDiff.y);
+                const auto distance = std::max(distDiff.x, distDiff.y);
```

Comment:
Chebyshev distance

Reply:
Do we really need two separate ways to do distance?

Reply:
At least while we want to retain replay compatibility with vanilla, I imagine.